rest/persistent_config_test.go

```diff
@@ -161,6 +161,55 @@ func TestAutomaticConfigUpgradeError(t *testing.T) {
 	}
 }
 
+func TestUnmarshalBrokenConfig(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("CBS required")
+	}
+
+	tb := base.GetTestBucket(t)
+	defer tb.Close()
+
+	rt := NewRestTester(t, &RestTesterConfig{PersistentConfig: true})
+	defer rt.Close()
+	resp := rt.SendAdminRequest(http.MethodPut, "/newdb/",
+		fmt.Sprintf(
+			`{"bucket": "%s", "num_index_replicas": 0, "enable_shared_bucket_access": %t, "use_views": %t}`,
+			tb.GetName(), base.TestUseXattrs(), base.TestsDisableGSI(),
+		),
+	)
+
+	cnf := make(map[string]interface{}, 1)
+	_, err := rt.ServerContext().BootstrapContext.Connection.GetConfig(tb.GetName(), rt.ServerContext().Config.Bootstrap.ConfigGroupID, &cnf)
+	require.NoError(t, err)
+
+	// Add invalid json fields to the config
+	cnf["num_index_replicas"] = "0"
+	marshaled, err := json.Marshal(cnf)
+
+	// Set the config to _sync:dbconfig:default
+	err = tb.SetRaw("_sync:dbconfig:default", 0, nil, marshaled)
+	require.NoError(t, err)
```

Comment:
I'm not 100% sure, but this SetRaw step might not be needed if your `UpdateConfig` callback is returning the broken marshalled version below to do the update.